tests/conftest.py

```diff
@@ -61,3 +61,7 @@ def clear_cuda_version():
     from conda.plugins.virtual_packages import cuda
 
     cuda.cached_cuda_version.cache_clear()
+
+
+# Do not register environments created during tests
+MonkeyPatch().setenv("CONDA_REGISTER_ENVS", "false")
```

Comment:
Let's use an autouse fixture here instead of running the monkey patcher manually during config module import time. This would mess up the order of how pytest manages monkey patches.

```suggestion
@pytest.fixture(autouse=True)
def do_not_register_envs(monkeypatch):
    """Do not register environments created during tests"""
    monkeypatch.setenv("CONDA_REGISTER_ENVS", "false")
```